src/components/controls/TextArea/TextAreaControl.tsx

```diff
@@ -54,46 +40,26 @@ export function TextAreaControl(props: Props) {
     } = props;
     const innerControlRef = React.useRef<HTMLTextAreaElement>(null);
     const handleRef = useForkRef(controlRef, innerControlRef);
-    const textareaRows = rows || minRows;
-    const innerValue = value || innerControlRef?.current?.value;
 
-    const resizeHeight = React.useCallback(() => {
-        const control = innerControlRef?.current;
+    const innerValue = value || innerControlRef?.current?.value;
 
-        if (control && !rows) {
-            const controlStyles = getComputedStyle(control);
-            const lineHeight = parseInt(controlStyles.getPropertyValue('line-height'), 10);
-            const paddingTop = parseInt(controlStyles.getPropertyValue('padding-top'), 10);
-            const paddingBottom = parseInt(controlStyles.getPropertyValue('padding-bottom'), 10);
-            const linesWithCarriageReturn = (innerValue?.match(/\n/g) || []).length + 1;
-            const linesByScrollHeight = calculateLinesByScrollHeight({
-                height: control.scrollHeight,
-                paddingTop,
-                paddingBottom,
-                lineHeight,
-            });
+    const realRows = React.useMemo(() => {
+        if (rows) return rows;
 
-            control.style.height = 'auto';
+        let currentRowsNumber = (innerValue?.match(/\n/g) || []).length + 1;
```

Comment:
It does not work in case of long strings without `\n`

Reply:
you are right, better calculate rows number depend on height

[d370a17](https://github.com/gravity-ui/uikit/pull/1714/commits/d370a17e942669816fa5c67ece2b634d1e502efa)